SU-HSDO/suhumsci
# STN-589: Hero Animation Refinement

# [STN-589](https://sparkbox.atlassian.net/browse/STN-589) READY FOR REVIEW

## Summary
Update when the animation is triggered on scroll, especially when multiple heros are on a page, in the components region.

Acceptance Criteria:

* The hero located at the top of the page should always animate on page load.
* The first hero within the main content show not animate if the top edge (~ 20 to 25% of the hero’s height) is visible. It should animate once the user scrolls past the top 20 to 25% of hero.
* Check to verify the carousel continues to work as expected (it should not animate at this time).

## Need Review By (Date)
asap

## Urgency
medium-hot

## Steps to Test
1. In your CLI, run `npm run test` and confirm our tests pass.
2. After running `npm run build` to build the styles and JS locally, I recommend clearing the cache by running `lando drush @sparkbox_sandbox.local cr`. This will ensure you are seeing the latest styles and JS updates.
3. Go to the [Hero Height Refinements](http://sparkbox-sandbox.suhumsci.loc/hero-height-refinements) page located under QA and confirm the following for both the Colorful and Traditional themes:
    - [x] The hero at the top of the page animates on page load.
    - [x] The first hero within the main content does not animate when top edge of the hero displays in the browser window.

    Example of hero within viewport where the animation has not triggered yet.

    <img width="1172" alt="hero updates" src="https://user-images.githubusercontent.com/12678977/104045936-4db79100-51ad-11eb-8bd6-2ce4d9fcfac4.png">

4. Go to the [Carousel as hero](http://sparkbox-sandbox.suhumsci.loc/qa/carousel-hero-image) image page under QA. The hero and carousel styles are closing linked together. For this reason we should make sure that any changes to the hero styles do not change the carousel styles.
    - [x] Confirm the Carousel styles have not changed and it displays as expected. There are no animations for the carousel and all elements should be visible. Note that there is a hero in the main content of this page. You can see that it animates after beginning to scroll down the page as well.
